### PR TITLE
add test for use-infinite-scroll

### LINKED ISSUE #1066: Enhance Test Coverage for `@yamada-ui/use-infinite-scroll`
package: @yamada-ui/use-infinite-scroll

### Description
`@yamada-ui/use-infinite-scroll` has a test coverage of less than 90%. We will enhance it to above 90%.

### Target files
- [packages/hooks/use-infinite-scroll/src/index.ts](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/hooks/use-infinite-scroll/src/index.ts)
  - [L11](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/hooks/use-infinite-scroll/src/index.ts#L11)
  - [L23](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/hooks/use-infinite-scroll/src/index.ts#L23)
  - [L53](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/hooks/use-infinite-scroll/src/index.ts#L53)
  - [L174](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/hooks/use-infinite-scroll/src/index.ts#L174)
  - [L197](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/hooks/use-infinite-scroll/src/index.ts#L197)
  - [L207](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/hooks/use-infinite-scroll/src/index.ts#L207)
  - [L241](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/hooks/use-infinite-scroll/src/index.ts#L241)
  - [L256](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/hooks/use-infinite-scroll/src/index.ts#L256)
  - [L268](https://app.codecov.io/gh/yamada-ui/yamada-ui/blob/main/packages/hooks/use-infinite-scroll/src/index.ts#L268)

### Problem Statement/Justification
Testing is essential to ensure quality when operating a package.

### Additional Information
While it's ideal to cover all tests, if there are too many lines to work on, please create a PR within your capacity.
Any deficiencies not covered by the PR will be automatically created as issues.

### PR DESCRIPTION
**Pull request by suzukisan22**
Closes #1066

## Description
- add test for use-infinite-scroll
- some of coverage can't cover because IntersectionObserver has to be mocked so can't test callback function.

## Is this a breaking change (Yes/No):

- No

### CODE REVIEW

File: packages/components/infinite-scroll-area/tests/infinite-scroll-area.test.tsx [hunk 1]
```diff
@@ -41,4 +33,76 @@ describe("<InfiniteScrollArea />", () => {
...
+      )
+    }
+
+    const { container, rerender } = render(<MyComponent isReverse={true} />)
+    await waitFor(async () => {
+      fireEvent.scroll(container, {
+        target: {
+          scrollTop: 1000,
+        },
+      })
```
Comment (108yen) [inline packages/components/infinite-scroll-area/tests/infinite-scroll-area.test.tsx:93]: Do not fire events in `waitFor`.

**Review by 108yen — CHANGES_REQUESTED** [review]

**Comment by 108yen** [comment]
Check Quality

File: packages/components/infinite-scroll-area/tests/infinite-scroll-area.test.tsx [hunk 2]
```diff
@@ -41,4 +33,76 @@ describe("<InfiniteScrollArea />", () => {
...
+
+    const { container, rerender } = render(<MyComponent isReverse={true} />)
+    fireEvent.scroll(container, {
+      target: {
+        scrollTop: 1000,
+      },
+    })
+    await waitFor(async () => {
+      expect(container.scrollTop).toBe(1000)
+    })
```
Comment (108yen) [inline packages/components/infinite-scroll-area/tests/infinite-scroll-area.test.tsx:95]: Unnecessary `async`
```ts
    await waitFor(() => expect(container.scrollTop).toBe(1000))
```

File: packages/components/infinite-scroll-area/tests/infinite-scroll-area.test.tsx [hunk 3]
```diff
@@ -41,4 +33,76 @@ describe("<InfiniteScrollArea />", () => {
...
+
+    rerender(<MyComponent isReverse={true} />)
```
Comment (108yen) [inline packages/components/infinite-scroll-area/tests/infinite-scroll-area.test.tsx:97]: Is this necessary?

File: packages/components/infinite-scroll-area/tests/infinite-scroll-area.test.tsx (hunk 2 above)
Comment (suzukisan22) [reply packages/components/infinite-scroll-area/tests/infinite-scroll-area.test.tsx:95]: but `await` is wait to get promise return. so if remove async, no need await.

Comment (108yen) [reply packages/components/infinite-scroll-area/tests/infinite-scroll-area.test.tsx]: This is not asynchronous, so `async` is meaningless.
```ts
expect(container.scrollTop).toBe(1000)
```
Assertions within `waitFor` will be executed multiple times until they succeed.

**Review by 108yen — APPROVED** [review]

**Review by hirotomoyamada — APPROVED** [review]

**Pull request merged**